src/components/Editor/index.js

```diff
@@ -143,9 +143,16 @@ class Editor extends PureComponent<Props, State> {
       return;
     }
 
-    this.setText(nextProps);
-    this.setSize(nextProps);
-    this.scrollToLocation(nextProps);
+    showLoading(this.state.editor);
```

Comment:
Maybe we could do a quick check for the size of the source and avoid flashing the loading state if it is "small".